packages/azure/src/azure-openai-provider.ts

```diff
@@ -101,8 +107,27 @@ export function createAzure(
       description: 'Azure OpenAI resource name',
     });
 
+  const checkIfBaseUrlIsProvided = () =>
+    loadSetting({
+      settingValue: options.baseURL,
+      settingName: 'baseURL',
+      environmentVariableName: 'AZURE_BASE_URL',
+      description: 'Azure OpenAI base URL',
+      failOnMissing: false,
+    });
```

Comment:
I would much prefer if we can only pass in the base URL through options for now (no env param loading). This means all the "failOnMissing" etc is not required.

The env loading is reserved for only the super-common options to get started. The expectation is to use options once the configuration gets more complex.

Reply:
(we need to document, test, and maintain this code and my preference is to keep it as simple as possible)

Reply:
Alright, we can do that! I agree that this is simpler and less prone.

Reply:
I could also remove the small test I added for the provider to check that the `baseURL` param works. Should we keep that or also remove it if we say that `baseURL` is not a core parameter?

Reply:
the test is great! I'll continue on the PR tomorrow and add docs & changelog. Thank you!